engines/ags/ags.cpp

```diff
@@ -155,14 +156,18 @@ Common::Error AGSEngine::run() {
 
 	setDebugger(new AGSConsole(this));
 
-	const char *filename = _gameDescription->desc.filesDescriptions[0].fileName;
-	if (_gameDescription->features & GAMEFLAG_INSTALLER) {
+	Common::String filename(_gameDescription->desc.filesDescriptions[0].fileName);
```

Comment:
Hm, I wonder if all of this could/should be generalized within AdvancedDetector. Something like loadArchivesFromDetectionEntry(int priority) which would scan the file list and add archives to SearchMan.

Reply:
Though you would need to extract the filename still to load it into argv further down... maybe the function could return a StringList of detected archive names? Honestly not sure if that would be overkill.

Reply:
In that case, it might be best to deal with this in a separate PR.